### ISSUE TITLE
One source of "jerkyness" may have been identified

### ISSUE THREAD

**Issue by tamlin-mike** [OpenXRay/xray-16#142]
Repro: Start a new game, run 30-or-so m.

Starting a new game in a debug build, while running under a profiler, I found out some lua script loaded a bunch of audio files - most if not all completely unrelated to the area I entered, and definitely not somethign expected to be played within the next seconds.

This made me realize the audio subsystem needs at least one, and maybe two, new functions.
1. preload.
   This one should be obvious. What may not be obvious is that it should run in another thread (think WorkItem's).
2. play_ambience.
   If added, this one should use the same preload (or even WorkQueue) mechanism, to not lock up the main game pointlessly.

**Comment by revolucas** [comment]
I suppose you mean **sound_theme.script**? It loads all the character_voices defined in **script_sound.ltx** when a NPC comes online; assuming it matches the community for **avail_communities** field. Probably, this script needs rewritten to account for story_id, so that regular stalkers aren't needlessly setting up to play quest sounds.

**Comment by Xottab-DUTY** [comment]
Solved with preload on precaching stage. (#407, da654d6c5ed221ab8762ab3991408bb8afd9d8d0 and a4920069f57ab1bca59800d41a2feaba4c80eb92)

**Comment by Xottab-DUTY** [comment]
~~Duplicate of #62~~ Oh, no, it's not duplicate.

Related to #62

**Comment by Xottab-DUTY** [comment]
Recently I've added parallel sound data preloading during the work on SteamAudio.

Now it makes sense to rework it more:
- [ ] Create sound sources in parallel on demand instead of precaching all sound sources at once (as it was introduced in #407).
This will reduce the RAM usage and speed-up game loading.

FYI: sound data is the actual PCM data from already created sound source, so sound source is obviously a different thing.
